springboot/src/main/java/ca/jrvs/apps/trading/service/TraderAccountService.java

```diff
@@ -0,0 +1,172 @@
+package ca.jrvs.apps.trading.service;
+
+import ca.jrvs.apps.trading.dao.AccountDao;
+import ca.jrvs.apps.trading.dao.PositionDao;
+import ca.jrvs.apps.trading.dao.SecurityOrderDao;
+import ca.jrvs.apps.trading.dao.TraderDao;
+import ca.jrvs.apps.trading.model.domain.Account;
+import ca.jrvs.apps.trading.model.domain.Position;
+import ca.jrvs.apps.trading.model.domain.SecurityOrder;
+import ca.jrvs.apps.trading.model.domain.Trader;
+import ca.jrvs.apps.trading.model.view.TraderAccountView;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import java.util.Optional;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+
+import javax.persistence.EntityNotFoundException;
+
+@Service
+public class TraderAccountService {
+
+    private TraderDao traderDao;
+    private AccountDao accountDao;
+    private SecurityOrderDao securityOrderDao;
+    private PositionDao positionDao;
+
+    @Autowired
+    public TraderAccountService(TraderDao traderDao, AccountDao accountDao,
+                                SecurityOrderDao securityOrderDao, PositionDao positionDao){
+        this.traderDao = traderDao;
+        this.accountDao = accountDao;
+        this.securityOrderDao = securityOrderDao;
+        this.positionDao = positionDao;
+    }
+
+
+    public TraderAccountView createTraderAndAccount(Trader trader) throws IllegalArgumentException{
+        if (trader == null){ throw new IllegalArgumentException("The trader cannot be null.");}
```

Comment:
use your IDE to autofromat

Reply:
use OR 
e.g. `if (trader.getCountry() == null || trader.getDob() == null `

Reply:
@jarviscanada But then I cannot give a specific exception to give info on which parameter is not valid?